mixer/cmd/reset.go

```diff
@@ -0,0 +1,205 @@
+		stateMixPreviousMixVersionUint32, err = parseUint32(b.State.Mix.PreviousMixVer)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Make sure FORMAT in mixer.state has the same value as update/www/<previousMixVer>/format
+	var format []byte
+	filename := filepath.Join(b.Config.Builder.ServerStateDir, "www", b.State.Mix.PreviousMixVer, "format")
+	if format, err = ioutil.ReadFile(filename); err != nil {
+		return err
+	}
+	b.State.Mix.Format = strings.TrimSpace(string(format))
+
+	// Change upstreamVersion file content to the previous version
+	var lastStableMixUpstreamVersion []byte
+	filename = filepath.Join(b.Config.Builder.ServerStateDir, "www", b.State.Mix.PreviousMixVer, "upstreamver")
+	if lastStableMixUpstreamVersion, err = ioutil.ReadFile(filename); err != nil {
+		return err
+	}
+	if strings.TrimSpace(string(lastStableMixUpstreamVersion)) != b.UpstreamVer {
+		// Set the upstream version to the previous format's latest version
+		b.UpstreamVer = strings.TrimSpace(string(lastStableMixUpstreamVersion))
+		b.UpstreamVerUint32, err = parseUint32(b.UpstreamVer)
+		if err != nil {
+			return errors.Wrapf(err, "Couldn't parse upstream version")
+		}
+		vFile := filepath.Join(b.Config.Builder.VersionPath, b.UpstreamVerFile)
+		if err = ioutil.WriteFile(vFile, []byte(b.UpstreamVer), 0644); err != nil {
+			return err
+		}
+	}
+
+	// Change mixVersion to point to PreviousMixVer
+	if err := ioutil.WriteFile(filepath.Join(b.Config.Builder.VersionPath, b.MixVerFile), []byte(b.State.Mix.PreviousMixVer), 0644); err != nil {
+		return err
+	}
+	b.MixVer = b.State.Mix.PreviousMixVer
+	b.MixVerUint32, err = parseUint32(b.MixVer)
+	if err != nil {
+		return errors.Wrapf(err, "Couldn't parse mixer version")
+	}
+
+	// Make sure update/image/LAST_VER points to PreviousMixVer
+	if err = ioutil.WriteFile(filepath.Join(b.Config.Builder.ServerStateDir, "image", "LAST_VER"), []byte(b.State.Mix.PreviousMixVer), 0644); err != nil {
+		return fmt.Errorf("couldn't update LAST_VER file: %s", err)
+	}
+
+	// update the state.mix file
+	err = b.State.Save()
+	if err != nil {
+		return fmt.Errorf("couldn't update mixer.state file: %s", err)
+	}
+
+	// if clean flag not set
+	if !resetCmdFlags.clean {
+		return nil
+	}
+
+	// Remove any folder inside update/image for versions above prevMixVer
+	files, err := ioutil.ReadDir(b.Config.Builder.ServerStateDir + "/image")
+	if err != nil {
+		log.Println(err)
+	}
+
+	for _, f := range files {
+		if f.Name() == "LAST_VER" {
+			continue
+		}
+		dirNameInt, err := parseUint32(f.Name())
+		if err != nil {
+			log.Println(err)
+			continue
+		}
+		if dirNameInt > stateMixPreviousMixVersionUint32 {
+			err := os.RemoveAll(b.Config.Builder.ServerStateDir + "/image/" + f.Name())
+			if err != nil {
+				log.Println(err)
+				continue
+			}
+		}
+	}
+
+	// Remove any folder inside update/www for versions above prevMixVer
+	files, err = ioutil.ReadDir(b.Config.Builder.ServerStateDir + "/www")
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Any reason why you use `log.Fatal` for update/www while you use `log.Println` for update/image ?

Reply:
sorry forgot to change that one, should be log.Println